File: sort/selection_sort/selection_sort.cpp
#include <iostream>
using namespace std;

int main()
{
    int arr[50], size;
    cout << "Enter the size of array : " << endl;
    cin >> size ;
    cout << "Enter the elements: " << endl;
    // user-defined elements in the array
    for (int i = 0; i < size; i++)
    {
        cin >> arr[i];
    }
    for (int i = 0; i < size-1; i++)
    {
        int min_index = i;
        for (int j = i + 1; j < size; j++)
        {
            if (arr[min_index] > arr[j])
            {
                min_index = j;
            }
        }
        int temp;
        temp = arr[i];
        arr[i] = arr[min_index];
        arr[min_index] = temp;
    }
    cout << "Sorted array is: " << endl;
    for (int k = 0; k < size; k++)
    {
        cout << " " << arr[k] ;
    }
}

File: sort/selection_sort/selection_sort.py
def selection_sort_func(lists):
    # n-1 th iterations
    for i in range(0, len(lists)-1):
        # Assume the 1st element to have the min. index
        minIndex = i
        # n-i iterations to get the min. index
        for j in range(i+1, len(lists)):
            if lists[minIndex] > lists[j]:
                minIndex = j
        lists[minIndex], lists[i] = lists[i], lists[minIndex]
    return lists    

# User defined length of the list
SIZE = input("Enter the number of elements ")
# Creating an empty list
LISTS = []
for k in range(0, SIZE):
    element = input('Enter the number: ')
    # appending each user defined value into the list
    LISTS.append(element)

print('Using Selection Sort ', selection_sort_func(LISTS))
